types/jest/jest-tests.ts

```diff
@@ -307,8 +307,8 @@ const mock7 = jest.fn((arg: number) => arg);
 const mock8: jest.Mock = jest.fn((arg: number) => arg);
 // $ExpectType Mock<Promise<boolean>, [number, string, {}, [], boolean]>
 const mock9 = jest.fn((a: number, _b: string, _c: {}, _iReallyDontCare: [], _makeItStop: boolean) => Promise.resolve(_makeItStop));
-// $ExpectType Mock<never, [never]>
-const mock10 = jest.fn((arg: never) => { throw new Error(arg); });
+// $ExpectType Mock<never, []>
+const mock10 = jest.fn(() => { throw new Error(); });
```

Comment:
could you add a test like: 
```typescript
interface TestFn {
  test(x: number): string
}
const mock12 = jest.fn<TestFn['test']>((x) => x + '');
```
It would prevent people from removing the optional implementation.